python/port/mod/time/modtime.cpp

```diff
@@ -1,17 +1,21 @@
 extern "C" {
 #include "modtime.h"
+#include <py/runtime.h>
 }
 #include <ion/timing.h>
 #include "../../helpers.h"
 #include <py/smallint.h>
-#include <py/runtime.h>
 
 mp_obj_t modtime_sleep(mp_obj_t seconds_o) {
 #if MICROPY_PY_BUILTINS_FLOAT
-  micropython_port_interruptible_msleep(1000 * mp_obj_get_float(seconds_o));
+  mp_float_t tts = 1000 * mp_obj_get_float(seconds_o);
 #else
-  micropython_port_interruptible_msleep(1000 * mp_obj_get_int(seconds_o));
+  mp_int_t tts = 1000 * mp_obj_get_int(seconds_o);
 #endif
+  if(tts < 0){
+    mp_raise_ValueError("sleep length must be non-negative");
```

Comment:
I think we could simply do `mp_raise_ValueError(nullptr);` since even CPython doesn't give an actual error message:
```
>>> sleep(-2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IOError: [Errno 22] Invalid argument
```

Reply:
Cpython raise this exact error when time < 0 indeed (https://github.com/python/cpython/blob/95a8a0e01d1f4f49ad6b4d4f2ae3bbf93c462013/Modules/timemodule.c#L337)